### ISSUE TITLE
vagrant-spk should attempt to autoupdate on use

### ISSUE THREAD

**Issue by phildini** [sandstorm-io/vagrant-spk#157]
As a developer, I worry that I'm missing updates to vagrant-spk when I use it. 

It would be great if [1] vagrant-spk tried to auto-update on use (with confirmation from the user) or [2] vagrant-spk let the user know there was an update available, with helpful instructions for upgrading.

**Comment by ocdtrekkie** [comment]
This is definitely going to be more important if we're going to be pushing people to use newer VMs as we go as well. I'm going to add this to the milestone for the next major release on the roadmap.

Is there a well known "easy" method of updating a Python script and all of it's dependent files? We also have to think about the fact that vagrant-spk also supports Mac OS and Windows, and with Windows packaging it as an EXE and a Windows installer is going to add some complexity to this.

**Comment by ocdtrekkie** [comment]
Additional design thoughts:
- Having a notification for updates when you run it is far more trivial, and should be the bare minimum we try for in 1.1.
- It's easy to do a `vagrant-spk check-update` or the like, but if we want to have it run automatically, we need to decide either what commands should include the update check, or how to ensure it doesn't check incessantly... people often run five or six vagrant-spk commands in short order when packaging apps, and we only need to check like once a day. So we probably need to store last time we checked for updates.
- We probably shouldn't delete stacks during a vagrant-spk update in case someone has a custom one or wants to keep a legacy one. However, we probably need to ensure we delete any remaining files from stacks we update.

**Comment by paulproteus** [comment]
Suggestion on an implementation approach: Download the list of git tags on this repo using this Atom feed: https://github.com/sandstorm-io/vagrant-spk/tags.atom

**Comment by ocdtrekkie** [comment]
@paulproteus Yeah, I'd need a easily-programmatically parse-able copy of the releases page. That seems pretty adequate.
